crates/floresta-chain/src/pruned_utreexo/partial_chain.rs

```diff
@@ -176,8 +177,11 @@ impl PartialChainStateInner {
         del_hashes: Vec<bitcoin::hashes::sha256::Hash>,
     ) -> Result<u32, BlockchainError> {
         let height = self.current_height + 1;
-
-        if let Err(BlockchainError::BlockValidation(e)) = self.validate_block(block, height, inputs)
+        let ancestor = self.get_ancestor(height)?;
+        let flags = self.get_validation_flags(height);
+        if let Err(BlockchainError::BlockValidation(e)) = self
+            .consensus
+            .validate_block(block, &ancestor, height, inputs, flags, true)
         {
```

Comment:
and in some internal calls of 'validate_block()' i had to make assumptions when its nice to verify the script